src/test/scala/io/iohk/ethereum/domain/BlockchainSpec.scala

```diff
@@ -214,7 +214,7 @@ class BlockchainSpec extends AnyFlatSpec with Matchers with ScalaCheckPropertyCh
   }
 
   it should "return correct best block number after saving and rolling back blocks" in new TestSetup {
-    forAll(posIntGen(min = 1, max = maxNumberBlocksToImport)) { numberBlocksToImport =>
+    forAll(intGen(min = 1, max = maxNumberBlocksToImport)) { numberBlocksToImport =>
```

Comment:
I did not understand why this test was failing in a previous PR, the generated number sometimes where negative so I added the `posIntGen` that had a clause filtering negative numbers.
In this branch this test was failing again sometimes so, looking into it, I realized that now that the cache was removed in previous PR's the way the value `expectedPersistedBestBlock` (line 257) was being calculated made no sense and simplified the test